source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -233,16 +243,21 @@ int DefaultCertValidator::doSynchronousVerifyCertChain(
                            ? validated != Envoy::Ssl::ClientValidationStatus::Failed
                            : validated == Envoy::Ssl::ClientValidationStatus::Validated;
 
-  return (allow_untrusted_certificate_ || success) ? 1 : 0;
+  return (allow_untrusted_certificate_ || success);
 }
 
-Envoy::Ssl::ClientValidationStatus DefaultCertValidator::verifyCertificate(
-    X509* cert, const std::vector<std::string>& verify_san_list,
-    const std::vector<SanMatcherPtr>& subject_alt_name_matchers) {
+Envoy::Ssl::ClientValidationStatus
+DefaultCertValidator::verifyCertificate(X509* cert, const std::vector<std::string>& verify_san_list,
+                                        const std::vector<SanMatcherPtr>& subject_alt_name_matchers,
+                                        std::string* error_details, uint8_t* out_alert) {
   Envoy::Ssl::ClientValidationStatus validated = Envoy::Ssl::ClientValidationStatus::NotValidated;
-
   if (!verify_san_list.empty()) {
     if (!verifySubjectAltName(cert, verify_san_list)) {
+      const char* error = "verify cert failed: verify SAN list";
+      if (error_details != nullptr) {
+        *error_details = error;
+      }
+      ENVOY_LOG(debug, error);
```

Comment:
Is it `error_details` since `error` is fixed?

Reply:
Using either `error` or `error_details` should be fine. What do you mean by `error` being fixed?

Reply:
I think `error` is not changed since:
```
      if (error_details != nullptr) {
        *error_details = error;
      }